lib/queue.js

```
@@ -1,33 +1,54 @@
+// last in in first out
 class Queue {
     constructor() {
-        // this.store = ...
-        throw new Error("This method has not been implemented!");
+        this.store = new Array(20);
+        this.front = 0;
+        this.rear = 0;
+        this.size = 0;
+        this.bufferSize = 20
     }
 
     enqueue(element) {
-        throw new Error("This method has not been implemented!");
-    }
+        // adds element to the end of the queue
+        
+        if (this.size === this.bufferSize){
+            return;
+        }
 
+        this.store[this.rear] = element;
+        this.rear = (this.rear + 1) % this.bufferSize;
+        this.size += 1;
+    }
+//removes and returns element form the front of queue and none if queue is empty
     dequeue() {
-        throw new Error("This method has not been implemented!");
+
+        if (this.front === this.rear) {
+            return;
```

👀 Again, we want to throw an error here letting the user know the queue is full